### PR TITLE
Alerting: Refactor fetch for receivers for using  RTKQuery

### PR DESCRIPTION
**Pull request by soniaAguilarPeiron**
**What this PR does / why we need it**:

This PR is a refactor for using [ReduxT oolkit Query](https://redux-toolkit.js.org/rtk-query/overview) when fetching receivers for getting contact points state.

### CODE REVIEW

File: public/app/features/alerting/unified/components/receivers/ReceiversTable.tsx [hunk 1]
```diff
@@ -122,8 +122,9 @@ function ReceiverHealth({ errorsByReceiver, someWithNoAttempt }: ReceiverHealthP
   );
 }
 const useContactPointsState = (alertManagerName: string) => {
-  const contactPointsStateRequest = useUnifiedAlertingSelector((state) => state.contactPointsState);
-  const { result: contactPointsState } = (alertManagerName && contactPointsStateRequest) || initialAsyncRequestState;
+  const { currentData: contactPointsState } = receiversApi.endpoints.contactPointsState.useQueryState({
```
Comment (konrad147) [inline public/app/features/alerting/unified/components/receivers/ReceiversTable.tsx:125]: Could we use `useContactPointsStateQuery` hook?

File: public/app/features/alerting/unified/api/receiversApi.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import { ContactPointsState } from 'app/types';
+
+import { alertingApi } from './alertingApi';
+import { fetchContactPointsState } from './grafana';
+
+export const receiversApi = alertingApi.injectEndpoints({
+  endpoints: (build) => ({
+    contactPointsState: build.query<ContactPointsState, { amSourceName: string }>({
```
Comment (konrad147) [inline public/app/features/alerting/unified/api/receiversApi.ts:8]: I think we could make it even simpler. Let's have a chat about it

File: public/app/features/alerting/unified/components/receivers/ReceiversTable.tsx (hunk 1 above)
Comment (soniaAguilarPeiron) [reply public/app/features/alerting/unified/components/receivers/ReceiversTable.tsx:125]: 💯

File: public/app/features/alerting/unified/state/actions.ts [hunk 3]
```diff
@@ -458,12 +458,6 @@ export const fetchGrafanaNotifiersAction = createAsyncThunk(
   (): Promise<NotifierDTO[]> => withSerializedError(fetchNotifiers())
 );
 
-export const fetchContactPointsStateAction = createAsyncThunk(
```
Comment (konrad147) [inline public/app/features/alerting/unified/state/actions.ts:461]: 👏

**Review by konrad147 — APPROVED** [review]
Nice! 🎉

**Pull request merged**